### PR TITLE
Support for writing DTD/XML declaration

### PR DESCRIPTION
**Pull request by HyukjinKwon**
https://github.com/databricks/spark-xml/issues/59
This PR implements the support for writing DTD and XML declaration as an option.
Simply this adds the given value. 
Also, while correcting this, I found a bug for reading attributes in an element in array.
This PR also corrects this.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
It looks there are too many options added. I think I should try to avoid adding options but I think this one is needed in order to write a well-form XML files.

**Comment by HyukjinKwon** [comment]
It looks there are too many options added and I think I should try to avoid adding options. However, I think this one is needed in order to write a well-form XML files.

File: src/main/scala/com/databricks/spark/xml/parsers/StaxXmlGenerator.scala [hunk 1]
```diff
@@ -40,7 +40,8 @@ private[xml] object StaxXmlGenerator {
       (vt, v) match {
         // If this is meant to be attribute, write an attribute
         case (_, null) | (NullType, _) if name.startsWith(options.attributePrefix) =>
-          writer.writeAttribute(name.substring(options.attributePrefix.size), options.nullValue)
+        // Because usually attributes having `null` do not exist in elements, just do not write
+        // attributes when given values are `null`.
```
Comment (HyukjinKwon) [inline src/main/scala/com/databricks/spark/xml/parsers/StaxXmlGenerator.scala]: I wonder if not writing attributes for null is appropriate. I think I should make a PR separately for this and the bug for reading attributes.

Comment (HyukjinKwon) [inline src/main/scala/com/databricks/spark/xml/parsers/StaxXmlGenerator.scala]: Fixed in #64.

File: src/main/scala/com/databricks/spark/xml/XmlOptions.scala [hunk 2]
```diff
@@ -52,6 +54,7 @@ private[xml] object XmlOptions {
     failFastFlag = parameters.get("failFast").map(_.toBoolean).getOrElse(false),
     attributePrefix = parameters.getOrElse("attributePrefix", DEFAULT_ATTRIBUTE_PREFIX),
     valueTag = parameters.getOrElse("valueTag", DEFAULT_VALUE_TAG),
-    nullValue = parameters.getOrElse("nullValue", DEFAULT_NULL_VALUE)
+    nullValue = parameters.getOrElse("nullValue", DEFAULT_NULL_VALUE),
+    headingContents = parameters.getOrElse("headingContents", DEFAULT_HEADING_CONTENTS)
```
Comment (HyukjinKwon) [inline src/main/scala/com/databricks/spark/xml/XmlOptions.scala]: Maybe I think I might have to verify the heading contents

File: src/main/scala/com/databricks/spark/xml/package.scala [hunk 3]
```diff
@@ -97,6 +97,7 @@ package object xml {
       val options = XmlOptions(parameters.toMap)
       val startElement = s"<${options.rootTag}>"
       val endElement = s"</${options.rootTag}>"
+      val headingContents = options.headingContents
```
Comment (HyukjinKwon) [inline src/main/scala/com/databricks/spark/xml/package.scala]: Maybe I think I might have to verify the heading contents
 Add a line note

**Comment by HyukjinKwon** [comment]
I will leave this open for this to find a better solution and also see if users need this.

**Comment by HyukjinKwon** [comment]
Let me close this for now and will open if it looks this PR is needed.

**Comment by paritosh-gupta** [comment]
Hey what do you think are the problems in this implementation? Would be nice to have DTD support for specially for legacy XML.

**Comment by HyukjinKwon** [comment]
@paritosh-gupta Actually, I have been thinking abiut this.. please feel free to open another issue and suggest ones if you have good ideas!

**Pull request closed without merging**